src/navsat_transform.cpp

```diff
@@ -310,6 +320,14 @@ namespace RobotLocalization
 
       utmWorldTransInverse_ = utmWorldTransform_.inverse();
 
+      // publish the origin of the world frame in NavSatFix format
+      sensor_msgs::NavSatFix nsf;
+      NavsatConversions::UTMtoLL(utmWorldTransInverse_.getOrigin().getY(), 
```

Comment:
Yeah.  I get it, and get why (and somewhat envy) it's more efficient.  But I'm a Python and JavaScript guy.  In JavaScript and some other languages you can actually return a reference to a value instantiated from within a function's scope.  It's called a closure and while it seems wrong to some and witchcraft to many, are very useful for asynchronous patterns and are what make singletons work well in JS.